### PR TITLE
win32: Fix buffer overflow in function readdir

### PR DESCRIPTION
**Pull request by remittor**

### CODE REVIEW

File: contrib/win32/win32compat/win32_dirent.c [hunk 1]
```diff
@@ -99,12 +99,12 @@ struct dirent *readdir(void *avp)
 			return NULL;
 		}
 
-		strncpy(pdirentry->d_name, tmp, strlen(tmp) + 1);
+		strncpy(pdirentry->d_name, tmp, sizeof(pdirentry->d_name));
```
Comment (bagajjal) [inline contrib/win32/win32compat/win32_dirent.c:102]: why will the tmp size is greater than d_name?
the file names are limited by MAX_PATH (260 characters)?

**Review by bagajjal — CHANGES_REQUESTED** [review]

File: contrib/win32/win32compat/win32_dirent.c (hunk 1 above)
Comment (remittor) [reply contrib/win32/win32compat/win32_dirent.c:102]: See **struct _wfinddata_t** => field **name** have 260 UTF16 symbols.
See **struct dirent** => field **d_name** have 256 bytes (UTF8).

`struct dirent {`
`	int            d_ino;       /* Inode number */`
`	char           d_name[256]; /* Null-terminated filename */`
`};`

Comment (bagajjal) [reply contrib/win32/win32compat/win32_dirent.c:102]: This is a bug which is fixed in the latest release..

Comment (bagajjal) [reply contrib/win32/win32compat/win32_dirent.c:102]: d_name in struct dirent is now of PATH_MAX size.

**Review by bagajjal — CHANGES_REQUESTED** [review]

**Pull request closed without merging**